fuel-relayer/src/relayer/syncing.rs

```diff
@@ -0,0 +1,70 @@
+use super::*;
+
+pub async fn wait_if_eth_syncing<P>(eth_node: &P) -> anyhow::Result<()>
+where
+    P: Middleware<Error = ProviderError> + 'static,
+{
+    let mut count = 0;
+    while get_status(eth_node).await? {
+        count += 1;
+        if count > 12 {
+            tracing::warn!(
+                "relayer has been waiting longer then a minute for the eth node to sync"
```

Comment:
And for example, if the `current_block` didn't change after the `X` synchronization rounds, we will return an error.

Reply:
I agree that a minute is too short, syncing a brand new eth node from scratch can even take up to several days!

Although I'd change the log message slightly since the relayer is actually waiting for the eth node to sync before it downloads logs for its own sync process:

```
tracing::info!("Waiting for the ethereum endpoint to finish Syncing: {:?}", status);
```

Reply:
Hmm, do we actually imagine we will be querying an eth node that is in the process of initial sync? Basically the relayer can't do anything while the eth node is syncing so a user might be confused why nothing is happening. I can definitely add a configurable reporting period.

Reply:
I have addressed this although I did not return an error if the current block hasn't changed because we don't want to panic the node over this.